vllm/model_executor/guided_decoding/reasoner/__init__.py

```diff
@@ -19,6 +19,10 @@ def get_reasoner(tokenizer: PreTrainedTokenizer,
         return None
     elif reasoning_backend == "deepseek_r1":
         return DeepSeekReasoner.from_tokenizer(tokenizer)
+    elif reasoning_backend == "granite":
+        logger.warning(
```

Comment:
Adding a warning for now since this is already a large PR, but I think adding a `GraniteReasoner` for guided decoding could be a follow-up later?

Reply:
SGTM!

Reply:
@aarnphm This will behave the same way as models with no reasoner. The intention here was mostly to clarify that there isn't a reasoning backend for granite in case users conflate it with `--enable-reasoning` / `--reasoning-parser granite` being supported for these models

Reply:
wfm.